app/src/main/java/me/gilo/wc/repo/CartRepository.kt

```diff
@@ -0,0 +1,96 @@
+package me.gilo.wc.repo
+
+
+import android.content.Context
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FirebaseFirestore
+import me.gilo.wc.common.CompletionGenericLiveData
+import me.gilo.wc.common.QueryLiveData
+import me.gilo.wc.common.WooLiveData
+import me.gilo.wc.models.CartLineItem
+import me.gilo.woodroid.Woocommerce
+import me.gilo.woodroid.models.LineItem
+import me.gilo.woodroid.models.Product
+import javax.inject.Inject
+
+open class CartRepository @Inject
+constructor() {
+
+    @Inject
+    lateinit var woocommerce: Woocommerce
+
+
+    private val cart: CollectionReference = FirebaseFirestore.getInstance()
+        .collection("users")
+        .document(FirebaseAuth.getInstance().currentUser!!.uid)
```

Comment:
Hello,

This piece of code crash when signin anonymously.

Reply:
I noticed that too, even in the old code. It's returning null probably because of user is not logged in yet. I will check on workaround for it

Reply:
`FirebaseAuth.getInstance().currentUser?.uid ?: "0"` should solve the crashing. `"0"` here is a default `uid` incase of null `user`

Reply:
Thx ! 

I also noticed :

java.lang.NumberFormatException: For input string: ""
at me.gilo.wc.ui.product.CartActivity.setUpPage(CartActivity.kt:147)

Maybe should check if regular_price is empty and use price instead.

And : "Converting to Arrays is not supported, please use Lists instead (found in field 'product.attributes.[0].options')"

Maybe use private ArrayList<String> options; in attribute3